tests/conftest.py

```diff
@@ -84,6 +109,22 @@ def fixture_sample2_data(sample_tag_names: List[str], second_sample_vcf: Path) -
     return {"tags": sample_tag_names, "file": second_sample_vcf}
 
 
+@pytest.fixture(scope="function", name="spring_file_1_with_tags")
+def fixture_spring_file_1_with_tags(
+    sample_id: str, spring_tag: str, spring_file_1: Path
+) -> dict:
```

Comment:
I think it can become messy with nested types within the dicts. And with `Any` the typing loses most of its advantages. Long and messy type hint is often a sign that creating a class might be good.